### DISCUSSION TITLE
Restore state from S3

### DISCUSSION THREAD

**Discussion by desprit** [openobserve/openobserve#2326] [General]
I'm deploying Openobserve in my kubernetes cluster and I think I successfully set up an S3 storage for it. At least, once the pod is alive I can see the following structure created in S3 bucket:
`files / myfolder / logs / logs / 2023 / 12 / 26 / 23` and many other folders. Every folder has a non-empty parquet file in it. 
But I tried restarting the pod and it doesn't seem to load old logs from s3. I can't find references in the docs suggesting if it's an expected behavior or not.

**Comment by hengfeiyang** [comment]
You need a permanent storage for store metadata, only the parquet files in s3, but we have metadata in local disk as default. it would help if you mounted a volume to `/data` directory or use etcd, postgres, mysql. etc..


https://openobserve.ai/docs/storage/
https://openobserve.ai/docs/ha_deployment/#metadata-store

**Reply by desprit** [reply]
Thank you, I didn't realize I need to store meta somewhere else. Will try that!
